### PR TITLE
"add_button" view for cardtype cards

### PR DESCRIPTION
**Pull request by xithan**

### CODE REVIEW

File: card/mod/01_core/set/all/collection.rb [hunk 1]
```diff
@@ -144,3 +144,47 @@ def set_search_params_variables! hash
...
+      tab_panels += tab_panel( id, tab_content, index == 0 )
+    end
+
+    wrap_with :div, :role=>"tabpanel" do
+      [
+        content_tag(:ul, tabs.html_safe, :class=>"nav #{args[:tab_type]}", :role=>"tablist"),
+        content_tag(:div, tab_panels.html_safe, :class=>'tab-content')
+      ]
+    end
+  end
```
Comment (ethn) [inline card/mod/01_core/set/all/collection.rb]: this is cool.  Perhaps we could optimize and make it so hidden tabs load later with ajax (to be more efficient)

File: card/mod/01_core/set/all/collection.rb [hunk 2]
```diff
@@ -144,3 +144,47 @@ def set_search_params_variables! hash
...
+    %{
+      <li role='presentation' #{'class="active"' if active}>
+        <a href=##{id} role='tab' data-toggle='tab'>#{name}</a>
+      </li>
+    }
+  end
+
+  def tab_panel id, content, active=false
+    %{
+      <div role='tabpanel' class="tab-pane #{'active' if active}" id=#{id}>
```
Comment (ethn) [inline card/mod/01_core/set/all/collection.rb]: is "pane" a typo or is that right?

File: card/mod/05_standard/set/type/cardtype.rb [hunk 3]
```diff
@@ -11,7 +11,22 @@
...
+      args[:params].gsub!(/=([^&]+)/) do |match|
+        "=#{$1.to_name.to_absolute context}"
+      end
+    end
+    %{
+      <a class='btn btn-default' href='/new/#{card.key}?#{args[:params]}'>
+        #{ _render_title args }
+      </a>
+    }
   end
```
Comment (ethn) [inline card/mod/05_standard/set/type/cardtype.rb]: I like it!  (Are we still thinking to add add_link view, too? (and, I guess, add_url, add_path...)

**Comment by ethn** [comment]
good to merge.  please respond to minor comments.

File: card/mod/01_core/set/all/collection.rb (hunk 2 above)
Comment (xithan) [inline card/mod/01_core/set/all/collection.rb]: "pane" is correct

File: card/mod/01_core/set/all/collection.rb (hunk 1 above)
Comment (xithan) [inline card/mod/01_core/set/all/collection.rb]: there is now a `tabs` and a `tabs_static` view

File: card/mod/05_standard/set/type/cardtype.rb (hunk 3 above)
Comment (xithan) [inline card/mod/05_standard/set/type/cardtype.rb]: added add_link, add_url, and add_path

**Comment by GerryG** [comment]
This looks pretty cool.  Is there a ticket that describes what it being done with this and related changes?  Seems like this is mostly about creating tab based views for card-lists which would be really great for navigation aids.

**Comment by ethn** [comment]
I don't think there is.  This was part of a quick fix for WikiRate.  Not sure we're ready to advertise it, but it's got a lot of potential!

**Comment by GerryG** [comment]
Yes, please make this a solid feature ASAP.  I've wanted tab controls based on lists for a long time.

Are there parts that you could start advertising a bit?

**Pull request merged**